editor/src/core/shared/import/import-operation-service.ts

```diff
@@ -0,0 +1,146 @@
+import { assertNever } from '../utils'
+import type { EditorAction, EditorDispatch } from '../../../components/editor/action-types'
+import {
+  setImportWizardOpen,
+  updateImportOperations,
+} from '../../../components/editor/actions/action-creators'
+import { ImportOperationAction } from './import-operation-types'
+import type {
+  ImportOperation,
+  ImportOperationResult,
+  ImportOperationType,
+} from './import-operation-types'
+import { isFeatureEnabled } from '../../../utils/feature-switches'
+
+let editorDispatch: EditorDispatch | null = null
```

Comment:
The dispatch function is available in the code almost everywhere, so I'd try and get rid of the global state here and check whether the dispatch function can be passed to functions in this file

Reply:
got it. is the preferred practice to pass it down? (asking since `fetchNodeModules`, for example, is being called from several places that don't have this `dispatch` function available)

Reply:
I'd prefer passing it down because I'd avoid global state as much as possible, but in terms of functionality if doesn't really matter, since the dispatch function always stays the same (afaik)

Reply:
Yes please, +1 to avoiding global state. If nothing else, it makes testing a nightmare

Reply:
fixed it here - https://github.com/concrete-utopia/utopia/pull/6507/commits/36cadbab90aef63b1b1c788ee73b3584fb24a8f8, it required a refactor of multiple files (some are unrelated to the actual functionality), but now the `dispatch` is passed down instead of being stored as a state